packages/babel-parser/src/parser/expression.js

```diff
@@ -748,7 +748,7 @@ export default class ExpressionParser extends LValParser {
     return this.finishNode(node, "TaggedTemplateExpression");
   }
 
-  atPossibleAsync(base: N.Expression): boolean {
+  atPossibleAsyncArrow(base: N.Expression): boolean {
     return (
```

Comment:
The conditions here corresponds to `atPossibleAsync`. I was meant to add `base.start === this.state.potentialArrowAt` here so it can be rebranded as `atPossibleAsyncArrow`.
```js
atPossibleAsyncArrow(base: N.Expression): boolean {
  return (base.type === "Identifier" &&
  base.name === "async" &&
  this.state.lastTokEnd === base.end &&
  !this.canInsertSemicolon() &&
  base.end - base.start === 5 &&
  base.start === this.state.potentialArrowAt)
}
```
The first 5 conditions are about `atPossibleAsync` and the last one is `hasPossibleArrow` so it combines into `atPossibleAsyncArrow`.

Reply:
I think I understand. I'll add the extra condition to this `atPossibleAsyncArrow` method, and delete the `atPossibleAsync` method. This means I will also change

```
if (!noCalls && this.atPossibleAsync(base)) {
```
in the Typescript plugin to
```
if (!noCalls && this.atPossibleAsyncArrow(base)) {
```

Reply:
If you are going with the `base.end - base.start === 5` route, could you add a comment in the code that it's needed to check that there are no escape sequences?

Reply:
Half joking: If `this.state.lastTokEnd >= base.end && base.start >= this.state.potentialArrowAt` is always true (I don't know). We can even further simplify it to
```js
return (base.type === "Identifier" &&
  base.name === "async" &&
  !this.canInsertSemicolon() &&
  this.state.lastTokEnd - this.state.potentialArrowAt === 5)
)
```